dataprofiler/tests/profilers/test_base_option.py

```diff
@@ -0,0 +1,54 @@
+import os
+import unittest
+from unittest import mock
+
+import pandas as pd
+
+from dataprofiler import Data, ProfilerOptions, Profiler
+from dataprofiler.profilers.profiler_options import BaseOption
+
+
+@mock.patch('dataprofiler.profilers.data_labeler_column_profile.'
+            'DataLabelerColumn.update', return_value=None)
+@mock.patch('dataprofiler.profilers.data_labeler_column_profile.DataLabeler')
+class TestBaseOption(unittest.TestCase):
+	
+	@classmethod
+	def setUpClass(cls):
+		cls.data = Data(data=pd.DataFrame([1, 2]), data_type='csv')
+	
+	def sanity(self, *mocks):
+		options = BaseOption()
+	
+	def test_set_helper(self, *mocks):
+		options = BaseOption()
+
+		#Options Is Not A Dictionary
+		expected_error = "The options must be a dictionary."
+		with self.assertRaisesRegex(ValueError, expected_error):
+			options._set_helper("notadictionary", "")
+		with self.assertRaisesRegex(ValueError, expected_error):
+			options._set_helper(["not", "a", "dictionary"], "")
+
+		#Variable Path Is Not A String
+		expected_error = "The variable path must be a string."
+		with self.assertRaisesRegex(ValueError, expected_error):
+			options._set_helper({"hello": "world"}, 1)
+		with self.assertRaisesRegex(ValueError, expected_error):
+			options._set_helper({}, 1)
+
+	def test_set(self, *mocks):
+		options = BaseOption()
+
+		#Options Is Not A Dictionary
+		expected_error = "The options must be a dictionary."
+		with self.assertRaisesRegex(ValueError, expected_error):
+			options.set("notadictionary")
+		with self.assertRaisesRegex(ValueError, expected_error):
+			options.set(["not", "a", "dictionary"])
+	
+	def test_validate_helper(self, *mocks):
+		pass
```

Comment:
are we gonna implement this test?

Reply:
Currently, the BaseOption class would raise a NotImplementedError for _validate_helper. I left the template here in case that changes in the future.

Reply:
@lettergram thoughts? I would suggest instead, we could validate that since this is abstract that we could validate that running the function would raise the `NotImplementedError`

i.e. 
```python
with self.assertRaises(NotImplementedError):
    options._validate_helper()
```

Reply:
that way, if changes happened in the code, this would the raise an error that something was altered and the test would need to be updated with the code changes.

Reply:
^ Agreed, recommend doing this.

Reply:
That makes sense, I will add it in

Reply:
Included in new commit